fidget/src/core/context/mod.rs

```diff
@@ -637,6 +637,39 @@ impl Context {
         self.op_binary(a, b, BinaryOpcode::Mod)
     }
 
+    /// Builds a node that returns the first node if the condition is not
+    /// equal to zero, else returns the other node
+    ///
+    /// The result is `a` if `condition != 0`, else the result is `b`.
+    /// ```
+    /// # let mut ctx = fidget::context::Context::new();
+    /// let x = ctx.x();
+    /// let y = ctx.y();
+    /// let z = ctx.z();
+    ///
+    /// let if_else = ctx.if_nonzero_else(x, y, z).unwrap();
+    ///
+    /// assert_eq!(ctx.eval_xyz(if_else, 0.0, 2.0, 3.0).unwrap(), 3.0);
+    /// assert_eq!(ctx.eval_xyz(if_else, 1.0, 2.0, 3.0).unwrap(), 2.0);
+    /// assert_eq!(ctx.eval_xyz(if_else, 0.0, f64::NAN, 3.0).unwrap(), 3.0);
+    /// assert_eq!(ctx.eval_xyz(if_else, 1.0, 2.0, f64::NAN).unwrap(), 2.0);
+    /// ```
+    pub fn if_nonzero_else<Condition: IntoNode, A: IntoNode, B: IntoNode>(
+        &mut self,
+        condition: Condition,
+        a: A,
+        b: B,
+    ) -> Result<Node, Error> {
+        let condition = condition.into_node(self)?;
+        let a = a.into_node(self)?;
+        let b = b.into_node(self)?;
+
+        let lhs = self.and(condition, a)?;
+        let n_condition = self.not(condition)?;
+        let rhs = self.and(n_condition, b)?;
+        self.add(lhs, rhs)
```

Comment:
```suggestion
        self.or(lhs, rhs)
```
(otherwise, it won't be fully simplified after tracing evaluation)

Reply:
Yes, you are correct - I was thinking about this backwards, as you can tell from my comment in #58.